src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -5219,55 +5215,119 @@ private static boolean needIncrement(MutableBigInteger mdivisor, int roundingMod
         return commonNeedIncrement(roundingMode, qsign, cmpFracHalf, mq.isOdd());
     }
 
+    /**
+     * {@code FIVE_TO_2_TO[n] == 5^(2^n)}
+     */
+    private static final BigInteger[] FIVE_TO_2_TO = new BigInteger[20];
```

Comment:
As discussed before, the benchmarks show no significant performance regressions even for a million of trailing zeros if this cache is reduced to save resident memory footprint.
```suggestion
    private static final BigInteger[] FIVE_TO_2_TO = new BigInteger[16 + 1];
```